### PR TITLE
Redshift plugin 

### PR DESCRIPTION
**Pull request by sumitsum**
## Description 
1. Add new plugin to connect to AWS Redshift cluster and run query. 
2. Add unit test cases.

Fixes appsmithorg/appsmith #1267 

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
1. Tested the driver by connecting it to an instance of Redshift cluster running on cloud. All build TC passed using AWS redshift cred.
2. Created mock test cases since testcontainer support is not availabe. All mock TCs passed.
3. Tested with UI:
a. **Test datasource with valid credentials**
Screenshot 2020-12-16 at 12 09 35 PM

b. **Test datasource with invalid credentials**
Screenshot 2020-12-16 at 12 11 00 PM

c. **Select query**
Screenshot 2020-12-16 at 12 14 00 PM

d. **Get structure**
Screenshot 2020-12-16 at 12 15 31 PM

e. **Delete query**
Screenshot 2020-12-16 at 12 17 41 PM

f. **Insert query**
Screenshot 2020-12-16 at 12 22 32 PM

g. **Create query**
Screenshot 2020-12-16 at 12 26 12 PM

f. **New logo**
Screenshot 2020-12-16 at 5 04 17 PM

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/pom.xml [hunk 1]
```diff
@@ -14,7 +14,8 @@
     <packaging>pom</packaging>
 
     <modules>
-        <module>postgresPlugin</module>
+        <!--TODO: uncomment.-->
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/pom.xml:17]: Why comment out the other plugins? This is very dangerous. By mistake also if this slips through the cracks, there'll be lots of problems.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+        @Override
+        public Mono<ActionExecutionResult> execute(Connection connection,
+                                                   DatasourceConfiguration datasourceConfiguration,
+                                                   ActionConfiguration actionConfiguration) {
+
+            return (Mono<ActionExecutionResult>) Mono.fromCallable(() -> {
+
+                try {
+                    if (connection == null || connection.isClosed() || !connection.isValid(VALIDITY_CHECK_TIMEOUT)) {
+                        log.info("Encountered stale connection in Postgres plugin. Reporting back.");
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:119]: Please correct the plugin name in all log messages. Will cause issues during debugging.

File: app/server/appsmith-plugins/pom.xml (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/pom.xml]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 2 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:36]: Please don't use wildcard imports. Instead change the settings in intellij to disable this.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+                    if (!StringUtils.isEmpty(authentication.getDatabaseName())) {
+                        urlBuilder.append(authentication.getDatabaseName());
+                    }
+                }
+                url = urlBuilder.toString();
+            }
+
+            return Mono.fromCallable(() -> {
+                try {
+                    System.out.println(Thread.currentThread().getName() + ": Connecting to db");
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:284]:
```suggestion
                    System.out.println(Thread.currentThread().getName() + ": Connecting to Redshift db");
```

**Comment by trishaanand** [comment]
@sumitsum : For important logs that we do want to see, could we use System.out.println instead of log.error?

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 3 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:36]: Yes this is intelliJ issue. I need to fix it.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+                "         kcu.table_name,\n" +
+                "         kcu.ordinal_position;\n";
+
+        private Map<String, Object> getRow(ResultSet resultSet) throws SQLException {
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            int colCount = metaData.getColumnCount();
+            // Use `LinkedHashMap` here so that the column ordering is preserved in the response.
+            Map<String, Object> row = new LinkedHashMap<>(colCount);
+
+            for (int i = 1; i <= colCount; i++) {
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:125]: https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSetMetaData.html#getColumnType(int)

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+
+    public RedshiftPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Slf4j
+    @Extension
+    public static class RedshiftPluginExecutor implements PluginExecutor<Connection> {
+
+        private final Scheduler scheduler = Schedulers.boundedElastic();
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+                "          and rco.unique_constraint_name = rel_kcu.constraint_name\n" +
+                "          and kcu.ordinal_position = rel_kcu.ordinal_position\n" +
+                "where tco.constraint_type = 'FOREIGN KEY'\n" +
+                "order by kcu.table_schema,\n" +
+                "         kcu.table_name,\n" +
+                "         kcu.ordinal_position;\n";
+
+        private Map<String, Object> getRow(ResultSet resultSet) throws SQLException {
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            int colCount = metaData.getColumnCount();
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:134]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 8]
```diff
@@ -0,0 +1,553 @@
...
+        @Override
+        public Mono<Connection> datasourceCreate(DatasourceConfiguration datasourceConfiguration) {
+            try {
+                Class.forName(JDBC_DRIVER);
+            } catch (ClassNotFoundException e) {
+                return Mono.error(new AppsmithPluginException(AppsmithPluginError.PLUGIN_ERROR, "Error loading Redshift JDBC Driver class."));
+            }
+
+            String url;
+            AuthenticationDTO authentication = datasourceConfiguration.getAuthentication();
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 3 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 4 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 9]
```diff
@@ -0,0 +1,569 @@
...
+                "where tco.constraint_type = 'FOREIGN KEY'\n" +
+                "order by kcu.table_schema,\n" +
+                "         kcu.table_name,\n" +
+                "         kcu.ordinal_position;\n";
+
+        private Map<String, Object> getRow(ResultSet resultSet) throws SQLException, AppsmithPluginException {
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            if(metaData == null) {
+                System.out.println("Redshift plugin: getRow: metaData is null");
+                log.error("Redshift plugin: getRow: metaData is null");
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:130]: This line is not required.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 10]
```diff
@@ -0,0 +1,569 @@
...
+                            rowsList.add(row);
+                        }
+                    } else {
+                        rowsList.add(Map.of(
+                                "affectedRows",
+                                ObjectUtils.defaultIfNull(statement.getUpdateCount(), 0))
+                        );
+
+                    }
+                } catch (SQLException e) {
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:219]: Here you are only catching the SQLException, but getRow function throws this as well as AppsmithPluginException.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 11]
```diff
@@ -0,0 +1,569 @@
...
+        @Override
+        public Mono<Connection> datasourceCreate(DatasourceConfiguration datasourceConfiguration) {
+            try {
+                Class.forName(JDBC_DRIVER);
+            } catch (ClassNotFoundException e) {
+                return Mono.error(new AppsmithPluginException(AppsmithPluginError.PLUGIN_ERROR, "Error loading Redshift JDBC Driver class."));
+            }
+
+            String url;
+            DBAuth authentication = datasourceConfiguration.getAuthentication();
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:260]: I think when I suggested the code change, I also added a casting. Try to use `commit the suggestion` instead of copying it. This ensures that in most scenarios, there wouldn't be any copy paste error.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 12]
```diff
@@ -0,0 +1,569 @@
...
+            try {
+                if (connection == null || connection.isClosed() || !connection.isValid(VALIDITY_CHECK_TIMEOUT)) {
+                    log.info("Encountered stale connection in Redshift plugin. Reporting back.");
+                    throw new StaleConnectionException();
+                }
+            } catch (SQLException error) {
+                // This exception is thrown only when the timeout to `isValid` is negative. Since, that's not the case,
+                // here, this should never happen.
+                System.out.println("Error checking validity of Redshift connection." + error);
+                log.error("Error checking validity of Redshift connection.", error);
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:526]: Please remove this line since it is no longer required.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 13]
```diff
@@ -0,0 +1,553 @@
...
+                        } catch (SQLException e) {
+                            log.warn("Error closing Redshift Statement", e);
+                        }
+                    }
+                }
+
+                ActionExecutionResult result = new ActionExecutionResult();
+                result.setBody(objectMapper.valueToTree(rowsList));
+                result.setIsExecutionSuccess(true);
+                System.out.println(Thread.currentThread().getName() + ": In the RedshiftPlugin, got action execution result:"
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: I have removed the query result but kep the single line. It is helpful to know that the query has successfully executed.

File: app/server/appsmith-plugins/redshiftPlugin/src/test/java/com/external/plugins/RedshiftPluginTest.java [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+    public static void setUp() {
+        address = "address";
+        port = 5439;
+        username = "username";
+        password = "password";
+        dbName = "dbName";
+    }
+
+    private DatasourceConfiguration createDatasourceConfiguration() {
+        AuthenticationDTO authDTO = new AuthenticationDTO();
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/test/java/com/external/plugins/RedshiftPluginTest.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 9 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 10 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 11 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 12 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Hi @trishaanand , should I remove this line entirely ?

[the remark this answers is not in the document]
Comment (trishaanand) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: No Sumit. Would still keep this for some time to ensure that our threading schedulers work fine.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 15]
```diff
@@ -0,0 +1,567 @@
...
+                "where tco.constraint_type = 'FOREIGN KEY'\n" +
+                "order by kcu.table_schema,\n" +
+                "         kcu.table_name,\n" +
+                "         kcu.ordinal_position;\n";
+
+        private Map<String, Object> getRow(ResultSet resultSet) throws SQLException, AppsmithPluginException {
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            if(metaData == null) {
+                System.out.println("Redshift plugin: getRow: metaData is null");
+                throw new AppsmithPluginException(AppsmithPluginError.PLUGIN_ERROR, "metaData is null");
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:130]: This is the error that the developer user sees. Can we reword it to ensure that the developer user can use this to debug her query? Thanks!

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 16]
```diff
@@ -0,0 +1,567 @@
...
+                "          and kcu.ordinal_position = rel_kcu.ordinal_position\n" +
+                "where tco.constraint_type = 'FOREIGN KEY'\n" +
+                "order by kcu.table_schema,\n" +
+                "         kcu.table_name,\n" +
+                "         kcu.ordinal_position;\n";
+
+        private Map<String, Object> getRow(ResultSet resultSet) throws SQLException, AppsmithPluginException {
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            if(metaData == null) {
+                System.out.println("Redshift plugin: getRow: metaData is null");
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:129]: Can you also add the thread name at the beginning of the log? Just ensuring that the logs always get printed from the plugin with a thread name. Would make some of the thread level debugging easier some day in the future.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 17]
```diff
@@ -0,0 +1,567 @@
...
+                try {
+                    if (connection == null || connection.isClosed() || !connection.isValid(VALIDITY_CHECK_TIMEOUT)) {
+                        log.info("Encountered stale connection in Redshift plugin. Reporting back.");
+                        throw new StaleConnectionException();
+                    }
+                } catch (SQLException error) {
+                    // This exception is thrown only when the timeout to `isValid` is negative. Since, that's not the case,
+                    // here, this should never happen.
+                    System.out.println("Error checking validity of Redshift connection. " + error);
+                    log.error("Error checking validity of Redshift connection.", error);
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:187]: Please remove this line. Also what happens when the connection throws this SQL Exception? Should we continue forward? Should we cut our losses at this point? What should be the correct flow at this point?

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 18]
```diff
@@ -0,0 +1,567 @@
...
+                }
+
+                String query = actionConfiguration.getBody();
+
+                if (query == null) {
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:192]: Can we move this to the top of the function? We shouldn't even be doing the expensive connection validity check if we are going to return at this point.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 19]
```diff
@@ -0,0 +1,567 @@
...
+            }
+            if (authentication.getPassword() != null) {
+                properties.put(PASSWORD, authentication.getPassword());
+            }
+
+            if (CollectionUtils.isEmpty(datasourceConfiguration.getEndpoints())) {
+                url = datasourceConfiguration.getUrl();
+
+            } else {
+                StringBuilder urlBuilder = new StringBuilder("jdbc:redshift://");
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:280]: Move the string `jdbc:redshift://` to a private static field at the top of the class. We should go out of our way to ensure that we are not using strings inside our code.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 17 above)
Comment (trishaanand) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:187]: I think this error can never happen. Can we remove the log and instead return Mono.error which should get logged into sentry to ensure that if this case ever arises, we get a notification and check for the values of VALIDITY_CHECK_TIMEOUT and fix it ASAP?
Please also do this for get datasource structure function.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java [hunk 20]
```diff
@@ -0,0 +1,567 @@
...
+        public Mono<DatasourceStructure> getStructure(Connection connection, DatasourceConfiguration datasourceConfiguration) {
+            try {
+                if (connection == null || connection.isClosed() || !connection.isValid(VALIDITY_CHECK_TIMEOUT)) {
+                    log.info("Encountered stale connection in Redshift plugin. Reporting back.");
+                    throw new StaleConnectionException();
+                }
+            } catch (SQLException error) {
+                // This exception is thrown only when the timeout to `isValid` is negative. Since, that's not the case,
+                // here, this should never happen.
+                System.out.println("Error checking validity of Redshift connection." + error);
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:524]: Please check the comment in execute function around this.

File: app/server/appsmith-plugins/redshiftPlugin/src/test/java/com/external/plugins/RedshiftPluginTest.java [hunk 21]
```diff
@@ -0,0 +1,424 @@
...
+        endpoint.setPort(port.longValue());
+
+        DatasourceConfiguration dsConfig = new DatasourceConfiguration();
+        dsConfig.setAuthentication(authDTO);
+        dsConfig.setEndpoints(List.of(endpoint));
+        return dsConfig;
+    }
+
+    @Test
+    public void testDatasourceCreate() {
```
Comment (trishaanand) [inline app/server/appsmith-plugins/redshiftPlugin/src/test/java/com/external/plugins/RedshiftPluginTest.java:84]:
```suggestion
    public void testInvalidDatasourceCreate() {
```

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 18 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java:223]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: ok.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 15 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 16 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 17 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 19 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

File: app/server/appsmith-plugins/redshiftPlugin/src/test/java/com/external/plugins/RedshiftPluginTest.java (hunk 21 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/test/java/com/external/plugins/RedshiftPluginTest.java]: I have renamed it to testDatasourceCreateConnectionFailure() as it accurately describes the intended check i.e. to expect a failure exactly when the driver tries to establish a connection, but must execute rest of the code error free.

File: app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java (hunk 20 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/redshiftPlugin/src/main/java/com/external/plugins/RedshiftPlugin.java]: Done.

**Review by trishaanand — DISMISSED** [review]

**Review by trishaanand — APPROVED** [review]
